lsp-mode.el

```diff
@@ -2084,7 +2084,7 @@ WORKSPACE is the workspace that contains the diagnostics."
         (remhash file workspace-diagnostics)
       (puthash file (append diagnostics nil) workspace-diagnostics))
 
-    (run-hooks 'lsp-diagnostics-updated-hook)))
+    (run-hook-with-args 'lsp-diagnostics-updated-hook workspace)))
```

Comment:
Since you have changed this hook to have an argument, can you change its name to `lsp-diagnostics-updated-functions` instead? https://www.gnu.org/software/emacs/manual/html_node/elisp/Hooks.html

We need to declare the old hook as obsolete too.

Reply:
Changing this will break also the clients like lsp-treemacs. Can we let bind lsp--cur-workspace instead?

Reply:
Sure, didn't know about this convention.

Reply:
Check also if it is already bound (we bind it for the notifications so they are executed in the context of the source server).

Reply:
@yyoncho I preserved old hook, just made it obsolete. Should be fine?

> Check also if it is already bound (we bind it for the notifications so they are executed in the context of the source server).

Not sure what you mean here. That lsp--cur-workspace should be defined even if user switched buffer? Or I didn't get you?

Reply:
> Can we let bind lsp--cur-workspace instead?

Can do this if you think it's better. I'm not sure which approach would be preferable :)